src/openms/source/CHEMISTRY/ISOTOPEDISTRIBUTION/Container.cpp

```diff
@@ -0,0 +1,252 @@
+#include <OpenMS/CHEMISTRY/EmpiricalFormula.h>
+#include <OpenMS/CHEMISTRY/Element.h>
+
+#include <OpenMS/KERNEL/Peak1D.h>
+
+
+using namespace std;
+
+namespace OpenMS
+{
+
+  IsotopeDistribution::IsotopeDistribution()
+  {}
+
+  IsotopeDistribution::IsotopeDistribution(const IsotopeDistribution & isotope_distribution) :
+    distribution_(isotope_distribution.distribution_)
+  {}
+
+  IsotopeDistribution::~IsotopeDistribution()
+  {}
+
+  
+
+  IsotopeDistribution & IsotopeDistribution::operator=(const IsotopeDistribution & iso)
+  {
+    if (this != &iso)
+    {
+      distribution_ = iso.distribution_;
+    }
+    return *this;
+  }
+
+
+  void IsotopeDistribution::set(const ContainerType & distribution)
+  {
+    distribution_ = distribution;
+  }
+
+  const IsotopeDistribution::ContainerType & IsotopeDistribution::getContainer() const
+  {
+    return distribution_;
+  }
+
+  Peak1D::CoordinateType IsotopeDistribution::getMax() const
+  {
+    if (distribution_.empty())
+    {
+      return 0;
+    }
+    return distribution_[distribution_.size() - 1].getMZ();
+  }
+
+  Peak1D::CoordinateType IsotopeDistribution::getMin() const
+  {
+    if (distribution_.empty())
+    {
+      return 0;
+    }
+    return distribution_[0].getMZ();
+  }
+
+
+  Size IsotopeDistribution::size() const
+  {
+    return distribution_.size();
+  }
+
+  void IsotopeDistribution::clear()
+  {
+    distribution_.clear();
+  }
+
+  void IsotopeDistribution::trimIntensities(double cutoff)
+  {
+    distribution_.erase(
+      remove_if(distribution_.begin(),
+                distribution_.end(),
+                [&cutoff](const MassAbundance& sample)
+                {
+                  return sample.getIntensity() < cutoff;
+                }), distribution_.end());
+  }
+
+  void IsotopeDistribution::sort_(function<bool(const MassAbundance& p1, const MassAbundance& p2)> sorter)
+  {
+    sort(distribution_.begin(), distribution_.end(), sorter);
+  }
+
+  void IsotopeDistribution::sortByIntensity()
+  {
+    sort_([](const MassAbundance& p1, const MassAbundance& p2){
+        return p1.getIntensity() > p2.getIntensity();
+      });
+  }
+  
+  void IsotopeDistribution::transform_(function<void(MassAbundance&)> lambda)
+  {
+    for_each(distribution_.begin(), distribution_.end(), lambda);
+  }
+
+
+  IsotopeDistribution IsotopeDistribution::operator+(const IsotopeDistribution & iso) const
+  {
+    throw Exception::NotImplemented(__FILE__,__LINE__,OPENMS_PRETTY_FUNCTION);
+    return *this;
+  }
+
+  IsotopeDistribution & IsotopeDistribution::operator+=(const IsotopeDistribution & iso)
+  {
+    throw Exception::NotImplemented(__FILE__,__LINE__,OPENMS_PRETTY_FUNCTION);
+    return *this;
+  }
+
+  IsotopeDistribution & IsotopeDistribution::operator*=(Size factor)
+  {
+    throw Exception::NotImplemented(__FILE__,__LINE__,OPENMS_PRETTY_FUNCTION);
+    return *this;
+  }
+
+  IsotopeDistribution IsotopeDistribution::operator*(Size factor) const
```

Comment:
sound like something that can either be removed or moved to the derived class (probably also true for the operators above